ssl/src/main/java/org/wildfly/security/ssl/JDKSpecific.java

```diff
@@ -52,10 +121,26 @@ static void setHandshakeApplicationProtocolSelector(SSLEngine sslEngine, BiFunct
      */
 
     static String[] getApplicationProtocols(SSLParameters parameters) {
+        if (SSLPARAMETERS_GET_APPLICATION_PROTOCOLS != null) {
+            try {
+                return (String[]) SSLPARAMETERS_GET_APPLICATION_PROTOCOLS.invoke(parameters);
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                throw new UnsupportedOperationException(e);
+            }
+        }
+
         throw new UnsupportedOperationException();
     }
 
     static void setApplicationProtocols(SSLParameters parameters, String[] protocols) {
+        if (SSLPARAMETERS_SET_APPLICATION_PROTOCOLS != null) {
+            try {
+                SSLPARAMETERS_SET_APPLICATION_PROTOCOLS.invoke(parameters, (Object) protocols);
```

Comment:
Does this work or should it be `new Object[] {protocols}`?

Reply:
Yeah this works, I found the same in the Undertow code as well.  The main point was to stop the protocols array itself being seen as an array of arguments - when I test this Undertow is setting an array of three Strings.

Reply:
Right yeah. It seemed like it may work, I've just never tried it this way is all :)